File: java/Samples/BiometricStandards/CS/ANTemplateSampleCS/RecordCreateForms/ANType10RecordCreateForm.cs
using System;
using Neurotec.Biometrics.Standards;
using Neurotec.IO;
using Neurotec.Images;

namespace Neurotec.Samples.RecordCreateForms
{
	public partial class ANType10RecordCreateForm : ANRecordCreateForm
	{
		#region Public constructor

		public ANType10RecordCreateForm()
		{
			InitializeComponent();

			foreach (object value in Enum.GetValues(typeof(ANImageType)))
			{
				imageTypeComboBox.Items.Add(value);
			}
			imageTypeComboBox.SelectedIndex = 0;
		}

		#endregion

		#region Private properties

		private ANImageType ImageType
		{
			get
			{
				return (ANImageType)imageTypeComboBox.SelectedItem;
			}
		}

		private string Smt
		{
			get
			{
				return smtTextBox.Text;
			}
		}

		#endregion

		#region Protected methods

		protected override ANRecord OnCreateRecord(ANTemplate template)
		{
			ANType10Record record;
			if (imageLoader.CreateFromImage)
			{
				record = new ANType10Record(ANTemplate.VersionCurrent, Idc, ImageType, imageLoader.Src,
					imageLoader.ScaleUnits, imageLoader.CompressionAlgorithm, Smt, imageLoader.Image);
			}
			else if (imageLoader.CreateFromData)
			{
				using (NImage image = NImage.FromMemory(imageLoader.ImageData))
				{
					record = new ANType10Record(ANTemplate.VersionCurrent, Idc, ImageType, imageLoader.Src,
						imageLoader.ScaleUnits, imageLoader.CompressionAlgorithm, Smt, image)
						{
							HorzLineLength = imageLoader.Hll,
							VertLineLength = imageLoader.Vll,
							HorzPixelScale = imageLoader.Hps,
							VertPixelScale = imageLoader.Vps,
							ColorSpace = imageLoader.Colorspace
						};
				}
			}
			else
			{
				throw new NotSupportedException();
			}

			template.Records.Add(record);
			return record;
		}

		#endregion
	}
}

File: java/Samples/BiometricStandards/VB/ANTemplateSampleVB/RecordCreateForms/ANType10RecordCreateForm.vb
Imports System
Imports Neurotec.Biometrics.Standards
Imports Neurotec.IO
Imports Neurotec.Images

Namespace RecordCreateForms

	Partial Public Class ANType10RecordCreateForm
		Inherits ANRecordCreateForm
#Region "Public constructor"

		Public Sub New()
			InitializeComponent()

			For Each value As Object In System.Enum.GetValues(GetType(ANImageType))
				imageTypeComboBox.Items.Add(value)
			Next value
			imageTypeComboBox.SelectedIndex = 0
		End Sub

#End Region

#Region "Private properties"

		Private ReadOnly Property ImageType() As ANImageType
			Get
				Return CType(imageTypeComboBox.SelectedItem, ANImageType)
			End Get
		End Property

		Private ReadOnly Property Smt() As String
			Get
				Return smtTextBox.Text
			End Get
		End Property

#End Region

#Region "Protected methods"

		Protected Overrides Function OnCreateRecord(ByVal template As ANTemplate) As ANRecord
			Dim record As ANType10Record
			If imageLoader.CreateFromImage Then
				record = New ANType10Record(ANTemplate.VersionCurrent, Idc, ImageType, imageLoader.Src, imageLoader.ScaleUnits, imageLoader.CompressionAlgorithm, Smt, imageLoader.Image)
			ElseIf imageLoader.CreateFromData Then
				Using image As NImage = NImage.FromMemory(imageLoader.ImageData)
					record = New ANType10Record(ANTemplate.VersionCurrent, Idc, ImageType, imageLoader.Src, imageLoader.ScaleUnits, imageLoader.CompressionAlgorithm, Smt, image) With {.HorzLineLength = imageLoader.Hll, .VertLineLength = imageLoader.Vll, .HorzPixelScale = imageLoader.Hps, .VertPixelScale = imageLoader.Vps, .ColorSpace = imageLoader.Colorspace}
				End Using
			Else
				Throw New NotSupportedException()
			End If

			template.Records.Add(record)
			Return record
		End Function

#End Region
	End Class
End Namespace
